src/store/modules/DownloadModule.ts

```diff
@@ -23,34 +29,49 @@ export const DownloadModule = {
     namespaced: true,
 
     state: (): State => ({
-        allVersions: [],
-        assignId: 0,
-        downloadObject: null, // TODO: Check if the last element of allVersions can be used instead of this
+        allDownloads: [],
     }),
 
+    actions: <ActionTree<State, RootState>>{
+        addDownload({state}, initialMods: string[]): number {
+            const assignId = state.allDownloads.length;
+            const downloadObject: DownloadProgress = {
+                assignId,
+                initialMods,
+                modName: '',
+                progress: 0,
+                failed: false,
+            };
+            state.allDownloads = [...state.allDownloads, downloadObject];
+            return assignId;
+        },
+    },
+
     getters: <GetterTree<State, RootState>>{
         activeDownloadCount(state) {
-            const active = state.allVersions.filter(
-                dl => !dl[1].failed && dl[1].progress < 100
+            const active = state.allDownloads.filter(
+                dl => !dl.failed && dl.progress < 100
             );
             return active.length;
         },
+        currentDownload(state) {
+            return state.allDownloads[state.allDownloads.length-1] || null;
+        },
     },
 
     mutations: {
-        setDownloadObject(state: State, downloadObject: DownloadProgress) {
-            state.downloadObject = {...downloadObject};
-        },
-        pushDownloadObjectToAllVersions(state: State, params: { assignId: number, downloadObject: DownloadProgress }) {
-            state.allVersions = [...state.allVersions, [params.assignId, params.downloadObject]];
-        },
-        updateDownloadObject(state: State, params: { assignId: number, downloadVersion: [number, DownloadProgress]}) {
-            const newVersions = [...state.allVersions];
-            newVersions[params.assignId] = params.downloadVersion;
-            state.allVersions = newVersions;
+        updateDownload(state: State, update: UpdateObject) {
+            const newDownloads = [...state.allDownloads];
+            if (newDownloads[update.assignId].assignId !== update.assignId) {
+                throw new R2Error(
+                    'Failed to update download status.',
+                    `DownloadProgress with id of ${update.assignId} didn't have the correct assignId.`,
+                    'Try initiating the download again or restarting the app.'
+                );
+            } else {
```

Comment:
Nitpick: personal preference, but I see the `else` block a waste of lines and indentation when the `if` block always throws.